File: CodeSolutions/104. Maximum Depth of Binary Tree/104.cpp
# Language: Python
# Commit Timestamp: 2024-02-18T00:00:00-04:00
# Solution Logic: Placeholder for explanation.

# Language: Python
# Commit Timestamp: 2024-02-18T00:00:00-04:00
# Solution Logic: Placeholder for explanation.

# Language: Python
# Commit Timestamp: 2024-02-18T00:00:00-04:00
# Solution Logic: Placeholder for explanation.

class Solution {
 public:
  int maxDepth(TreeNode* root) {
    if (root == nullptr)
      return 0;
    return 1 + max(maxDepth(root->left), maxDepth(root->right));
  }
};

File: CodeSolutions/104. Maximum Depth of Binary Tree/104.py
# Language: Python
# Commit Timestamp: 2024-03-19T00:00:00-03:00
# Solution Logic: Placeholder for explanation.

# Language: Python
# Commit Timestamp: 2024-03-19T00:00:00-03:00
# Solution Logic: Placeholder for explanation.

# Language: Python
# Commit Timestamp: 2024-03-19T00:00:00-03:00
# Solution Logic: Placeholder for explanation.

class Solution:
  def maxDepth(self, root: TreeNode | None) -> int:
    if not root:
      return 0
    return 1 + max(self.maxDepth(root.left), self.maxDepth(root.right))
